File: C/Practice/calculator.c
#include <stdio.h>

int main () {
    double num1, num2;
    double result;
    char operation;

    printf("Enter First Number: ");
    scanf("%lf", &num1);

    printf("Please Enter and Operator (+ - * /): ");
    scanf(" %c", &operation);


    switch (operation)
    {
    case '+':
        printf("Enter Second Number: ");
        scanf("%lf", &num2);
        result = num1 + num2;
        printf("Here is the Result: %.2lf", result);
        break;
    
    case '-':
        printf("Enter Second Number: ");
        scanf("%lf", &num2);
        result = num1 - num2;
        printf("Here is the Result: %.2lf", result);
        break;
    case '*':
        printf("Enter Second Number: ");
        scanf("%lf", &num2);
        result = num1 * num2;
        printf("Here is the Result: %.2lf", result);
        break;
    case '/':
        printf("Enter Second Number: ");
        scanf("%lf", &num2);
        result = num1 / num2;
        printf("Here is the Result: %.2lf", result);
        break;
    
    default:
    printf("%c Is not a valid Operator", operation);
        break;
    }

    /*
    if (operation == '+') {
        printf("Enter Second Number: ");
        scanf("%lf", &num2);
        result = num1 + num2;
        printf("Here is the Result: %.2lf", result);
    
    }
    else if (operation == '-') {
        printf("Enter Second Number: ");
        scanf("%lf", &num2);
        result = num1 - num2;
        printf("Here is the Result: %.2lf", result);
    }
    else if (operation == '*') {
        printf("Enter Second Number: ");
        scanf("%lf", &num2);
        result = num1 * num2;
        printf("Here is the Result: %.2lf", result);
    }
        else if (operation == '/') {
        printf("Enter Second Number: ");
        scanf("%lf", &num2);
        result = num1 / num2;
        printf("Here is the Result: %.2lf", result);
    }
    else {
        printf("Invalid Operation! --> %c <--", operation);
        return 1;
    }*/

    return 0;
}

File: Java/Practice/src/Calculator.java
import java.util.Scanner;

public class Calculator {
    public static void main(String[] args)
    {
        Scanner input = new Scanner(System.in);

        do{
            
            double num1 = 0, num2 = 0, result = 0;
            char operat, response;
    
            System.out.print("\nEnter Num 1: ");
            num1 = input.nextDouble(); 
    
            System.out.print("Enter what operation to use (+ - * /): ");
            operat = input.next().charAt(0);

    
            System.out.print("Enter Num 2: ");
            num2 = input.nextDouble();
    
            switch (operat) {
                case '+':
                    result = num1 + num2;
                    System.out.println("\nResult: "+result);
                break;
                case '-':
                    result = num1 - num2;
                    System.out.println("\nResult: "+result);
    
                break;
                case '*':
                    result = num1 * num2;
                    System.out.println("\nResult: "+result);
                break;
                case '/':
                    result = (num2 != 0 ) ? num1 / num2 : 0;
                    System.out.println((result == 0) ? "Undefined" : "\nResult is: " + result);
                break;
            
                default:
                    System.out.println("Error");
                    break;
            }

                System.out.print("\nAgain? (y/n): ");
            response = input.next().toUpperCase().charAt(0);
            if(response == 'N' ){
                break;
            }
            else if (response != 'Y'){
                System.out.println("Invalid Response!");
                continue;
            }
        }while(true);
        System.out.println("GoodBye!");
        input.close();
        

    }
}
